packages/api/src/libs/utils.ts

```diff
@@ -40,12 +45,34 @@ export function calcGcr(
 }
 
 export function calcTvl(
+  // price needs to be in wei
   price: string,
   emp: Pick<uma.tables.emps.Data, "totalPositionCollateral" | "collateralDecimals">
 ) {
   const { totalPositionCollateral, collateralDecimals } = emp;
   assert(uma.utils.exists(totalPositionCollateral), "requires total position collateral");
   assert(uma.utils.exists(collateralDecimals), "requires collateralDecimals");
   const normalizedCollateral = uma.utils.ConvertDecimals(collateralDecimals, 18)(totalPositionCollateral);
-  return parseUnits(price).mul(normalizedCollateral).div(parseUnits("1"));
+  return BigNumber.from(price).mul(normalizedCollateral).div(ETH);
+}
+
+// collateralPrice is 1 collateral in USD, assumes price in wei
+// syntheticPrice is 1 synthetic to collateral, assumes price in wei
+export function calcSyntheticValue(
```

Comment:
this function needs some more eyes on it. This basically takes a synthetic price as returned from a bot price feed and a collateral price priced in USD, and comes up with a synthetic price in USD.  This was adapted from the affiliates logic. I guess the underlying assumption here is that a synthetic price as returned from the bot will always be against its collateral.